### DISCUSSION TITLE
How to use query caching

### DISCUSSION THREAD

**Discussion by rickstaa** [grafana/grafana-infinity-datasource#706] [Q&A] [answered]
I recently came across the addition of query caching to the documentation in [this commit](https://github.com/yesoreyeram/grafana-infinity-datasource/commit/f41ff39b6391b8945e36c6ec551309273ec7d607). However, I couldn't locate any documentation regarding the usage of this feature with your backend parser. Could you provide details on how to utilize query caching? Specifically, I'm interested in whether caching is supported through HTTP headers or if there is a separate setting within the data source that I can leverage. Any guidance on this would be greatly appreciated!

**Comment by yesoreyeram** [comment, marked as answer]
Query caching is a grafana enterprise feature.

**Reply by rickstaa** [reply]
Your prompt response is appreciated. While I had a suspicion along those lines, I was curious whether your plugin incorporates its caching solution, akin to the approach taken by https://grafana.com/grafana/plugins/marcusolsson-json-datasource 🤔.

**Reply by rickstaa** [reply]
@yesoreyeram would it, for example, be possible to allow this behaviour through the query headers?

![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/17570430/dafd8dad-a694-4aae-bbc9-d432589edd9c)

**Reply by yesoreyeram** [reply]
Unfortunately infinity plugin doesnt have any built-in support for caching. It just forward those headers to the target api. If your api supports caching, it does. But plugin itself doesn't hold anything.

**Reply by yesoreyeram** [reply]
Adding query caching was in my list. But due to existence of grafana enterprise query caching, i dropped the plan unfortunately.

**Reply by yesoreyeram** [reply]
Incorrectly setting caching in top of another caching layer may cause many side-effects

**Reply by rickstaa** [reply]
I got it! Thanks for the info. No worries at all. I'll use the https://grafana.com/grafana/plugins/marcusolsson-json-datasource/ plugin for data sources with rate limitations. 😊
